### PR TITLE
Use the c++11 standard, fix compiler warnings.

### PR DESCRIPTION
**Pull request by belovachap**
Changes needed to make the pars command line client compile after recent enhancements from @hrobeers 

I'm on linux mint 17, so my environment is: `g++ (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4` and `libboost1.55`

### CODE REVIEW

File: src/json/json_spirit_writer_template.h [hunk 1]
```diff
@@ -28,8 +28,6 @@ namespace json_spirit
     template< class String_type >
     String_type non_printable_to_string( unsigned int c )
     {
-        typedef typename String_type::value_type Char_type;
```
Comment (hrobeers) [inline src/json/json_spirit_writer_template.h]: Is this just an unused typedef?
Please leave it there as we don't want to diverge too much from upstream.

File: src/makefile.unix [hunk 2]
```diff
@@ -83,9 +83,21 @@ LIBS+= \
...
+    -Werror \
+    -Wno-sign-compare \
+    -Wno-invalid-offsetof \
+    -Wno-unused-parameter \
+    -Wformat \
+    -Wformat-security \
+    $(DEBUGFLAGS) \
+    $(DEFS) \
+    $(HARDENING) \
+    $(CXXFLAGS)
```
Comment (hrobeers) [inline src/makefile.unix]: Split the makefile changes into a separate commit.

File: src/walletdb.cpp [hunk 3]
```diff
@@ -389,41 +389,7 @@ void ThreadFlushWalletDB(void* parg)
...
-                } catch(const filesystem::filesystem_error &e) {
-                    printf("error copying wallet.dat to %s - %s\n", pathDest.string().c_str(), e.what());
-                    return false;
-                }
-            }
-        }
-        Sleep(100);
-    }
+    printf("BackupWallet disabled on account of filesystem::copy_file not compiling under c++11.");
+    printf("Please stop your client and manually backup your wallet file.");
```
Comment (hrobeers) [inline src/walletdb.cpp]: Disable this function in a separate commit, so we can easily revert the commit later.

Comment (belovachap) [inline src/walletdb.cpp]: will do :)

File: src/json/json_spirit_writer_template.h (hunk 1 above)
Comment (belovachap) [inline src/json/json_spirit_writer_template.h]: yeah, it's not used... might be hard to leave in since i added `-Werror` to the compile flags. don't really see why dropping it would be so bad; why don't we link to this library anyway? never understood why the bitcoin code base copied in versions of some libs...

Comment (hrobeers) [inline src/json/json_spirit_writer_template.h]: Ok, leave it in then

**Comment by belovachap** [comment]
@hrobeers split it up :)

**Comment by hrobeers** [comment]
Merged, but I had to fix another auto_ptr error that you missed, probably by not doing a full rebuild.

**Comment by belovachap** [comment]
:tada: :rainbow:
